File: rxbilling/src/main/java/com/vanniktech/rxbilling/PurchasedInApp.java
package com.vanniktech.rxbilling;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.google.auto.value.AutoValue;

@SuppressWarnings("PMD.UseObjectForClearerAPI") @AutoValue public abstract class PurchasedInApp implements Purchased {
  @NonNull public static PurchasedInApp create(@NonNull final String packageName, @NonNull final String productId, @NonNull final String purchaseToken, final int purchaseState, final long purchaseTime) {
    return create(packageName, productId, purchaseToken, purchaseState, purchaseTime, null);
  }

  @NonNull public static PurchasedInApp create(@NonNull final String packageName, @NonNull final String productId, @NonNull final String purchaseToken, final int purchaseState, final long purchaseTime, @Nullable final String orderId) {
    return new AutoValue_PurchasedInApp(packageName, productId, purchaseToken, purchaseState, purchaseTime, orderId);
  }
}

File: rxbilling/src/main/java/com/vanniktech/rxbilling/PurchasedInApp.kt
package com.vanniktech.rxbilling

data class PurchasedInApp(
  override val packageName: String,
  override val productId: String,
  override val purchaseToken: String,
  override val purchaseState: Int,
  override val purchaseTime: Long,
  override val orderId: String? = null,
) : Purchased
